### PR TITLE
chore: set max SDK version to <3.0.0

### PR DESCRIPTION
**Pull request by chalin**
cc @kwalrath @kevmoo

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -6,7 +6,9 @@ dart:
 
 # See https://docs.travis-ci.com/user/languages/dart/ for details.
 dart_task:
-  - test: --platform vm,firefox
+  - test: --platform vm
+    xvfb: false
```
Comment (nex3) [inline .travis.yml]: Why this change?

File: .travis.yml [hunk 2]
```diff
@@ -6,7 +6,9 @@ dart:
...
+  - test: --platform firefox
     install_dartium: true
```
Comment (nex3) [inline .travis.yml]: You shouldn't need to install Dartium anywhere anymore.

File: analysis_options.yaml [hunk 3]
```diff
@@ -1,2 +1 @@
 analyzer:
```
Comment (nex3) [inline analysis_options.yaml]: Again, just remove this file if it's not configuring anything.

File: .travis.yml (hunk 1 above)
Comment (chalin) [reply .travis.yml]: Splitting vm and browser tests makes it easier to know which platform has failures.

File: .travis.yml (hunk 2 above)
Comment (chalin) [reply .travis.yml]: But `glob` is still compatible with Dart 1: `sdk: '>=1.23.0 <3.0.0'`. That is why I left in `install_dartium: true`. Will tests run under Dart 1 w/o that line?

File: analysis_options.yaml (hunk 3 above)
Comment (chalin) [reply analysis_options.yaml]: Done

File: .travis.yml (hunk 1 above)
Comment (nex3) [reply .travis.yml]: Why are you disabling XVFB?

Splitting tests can be clearer, but it also takes more resources, which for small packages like this increases the iteration time and takes resources away from other dart-lang builds.

File: .travis.yml (hunk 2 above)
Comment (nex3) [reply .travis.yml]: You're not actually running tests on Dartium, so yes.

File: .travis.yml (hunk 1 above)
Comment (chalin) [reply .travis.yml]: Ok

File: .travis.yml (hunk 2 above)
Comment (chalin) [reply .travis.yml]: Removed

**Comment by chalin** [comment]
@nex3 or @kevmoo - could you restart the Travis jobs that failed (due to pub network issue)?

**Comment by kevmoo** [comment]
Done
